WalletWasabi/WabiSabi/Backend/Models/Alice.cs

```diff
@@ -25,14 +33,49 @@ public Alice(Coin coin, OwnershipProof ownershipProof)
 		public bool ConfirmedConnection { get; set; } = false;
 		public bool ReadyToSign { get; set; }
 
+		private CancellationTokenSource TimeoutCTS { get; } = new();
+
 		public long CalculateRemainingVsizeCredentials(int maxRegistrableSize) => maxRegistrableSize - TotalInputVsize;
 
 		public Money CalculateRemainingAmountCredentials(FeeRate feeRate) => Coin.EffectiveValue(feeRate);
 
-		public void SetDeadlineRelativeTo(TimeSpan connectionConfirmationTimeout)
+		public async void Dispose()
+		{
+			// Don't cancel the timeout when it is
+			using (await AsyncLock.LockAsync().ConfigureAwait(false))
```

Comment:
well the purpose of this is to wait until any other request on Alice has finished, but i think this is not necessary either way because of when `Dispose` can be invoked.